### PR TITLE
Dev 6

### PR DESCRIPTION
**Pull request by Denis-Balako**

### CODE REVIEW

File: src/main/java/com/balako/onlinebookstore/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        User user = userMapper.toModel(request);
+        user.setEmail(request.getEmail());
+        user.setPassword(passwordEncoder.encode(request.getPassword()));
+        user.setFirstName(request.getFirstName());
+        user.setLastName(request.getLastName());
+        user.setShippingAddress(request.getShippingAddress());
+        Role role = roleRepository.getByName(RoleName.ROLE_USER);
+        user.setRoles(Set.of(role));
+        User savedUser = userRepository.save(user);
+        return userMapper.toUserResponse(savedUser);
```
Comment (CodeWebWeaver) [inline src/main/java/com/balako/onlinebookstore/service/impl/UserServiceImpl.java:41]: You call 
User user = userMapper.toModel(request);
and then do mapper work. Why?

**Review by CodeWebWeaver — CHANGES_REQUESTED** [review]
This code is absolutely amazing. I haven't seen such beautiful code in my life. I want to cherish it and share it with my children someday. I want your autograph. But there is still one issue

**Review by CodeWebWeaver — APPROVED** [review]
Nice!

File: src/main/java/com/balako/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 2]
```diff
@@ -1,5 +1,6 @@
-package com.balako.onlinebookstore.exceptions;
+package com.balako.onlinebookstore.exception;
 
+import java.nio.file.AccessDeniedException;
 import java.time.LocalDateTime;
```
Comment (CodeWebWeaver) [inline src/main/java/com/balako/onlinebookstore/exception/CustomGlobalExceptionHandler.java:4]: Wrong import:
use springframework.security

**Review by CodeWebWeaver — CHANGES_REQUESTED** [review]

**Review by CodeWebWeaver — APPROVED** [review]
Good

File: src/main/java/com/balako/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 3]
```diff
@@ -35,6 +37,28 @@ protected ResponseEntity<Object> handleMethodArgumentNotValid(
         return new ResponseEntity<>(body, headers, status);
     }
 
+    @ExceptionHandler({RegistrationException.class})
+    protected ResponseEntity<Object> handleRegistrationException(
+            RegistrationException ex
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
```
Comment (vovasalyha) [inline src/main/java/com/balako/onlinebookstore/exception/CustomGlobalExceptionHandler.java:45]: You're creating the same map for each error response. I suggest to define the error response formate and create a separate class for error response.

File: src/main/java/com/balako/onlinebookstore/model/Role.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Entity
+public class Role {
```
Comment (vovasalyha) [inline src/main/java/com/balako/onlinebookstore/model/Role.java:17]: Table name?

**Review by vovasalyha — APPROVED** [review]
GJ. Take a look on few suggestions.

**Pull request merged**